helsingborg-stad/app-mitt-helsingborg
# feat: add EnvironmentService and EnvironmentServiceLocator

## Explain the changes you’ve made

Added new system for managing environment (the API endpoint-url and API key) in runtime instead of build-time.

## Explain why these changes are made

We will be moving from build-time environment configuration to runtime configuration in order to facilitate a monolithic build between development and production builds.

## Explain your solution

Added new EnvironmentService that will replace the current EnvironmentConfigurationService and use runtime-specified environments instead of reading from the build-time dotenv configuration.

Also added a EnvironmentServiceLocator using Service Locator pattern to handle EnvironmentService.

Changed files using EnvironmentConfigurationService to use new files instead.

EnvironmentConfigurationService (old) is not removed.

## How to test

Concrete example:

1. Checkout this branch
2. Add `MITTHELSINGBORG_IO` and `MITTHELSINGBORG_IO_APIKEY` to .env and set them to the default/fallback (probably same as existing `DEVELOP_MITTHELSINGBORG_IO_*`).
3. Run and verify it still works but not the default/fallback is always used regardless of dropdown-picker choice.

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.
